### PR TITLE
KA modkits in necropolis chests are now design discs to force miners to bring back minerals

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
tweak: KA modkits in necropolis chests are now design discs with designs for those modkits, to force miners to bring back minerals.
/:cl:

This should fix kor's concern that miners are getting too self-reliant and not bringing back enough minerals (see #27982 for context)

I didn't make the lifesteal crystal one a design though since it's literal magic

### CODE REVIEW

**Comment by DoctoreMad** [comment]
Wait those design disks
Does that mean that one disk drop equals infinite modkits?

**Comment by GunHog** [comment]
Agreed with this. Miners carry my favorite (and sometimes least favorite) interactions. Sometimes they are super nice and bring the minerals with them. Other times, they leave the minerals, ask for gear, and get mad when I tell them that I have no minerals (at which point I send someone to get them, or do it myself). In other times, miners break in and take over the R&D machines. 

Sometimes I try to offer a miner a mech and we have arguments over which mining method is better. All that to say that this keeps those miners coming to Sci so I can trade and argue with them!

**Comment by DoctoreMad** [comment]
Mining mechs are pretty shit though just make a Odysseus and be done with it.

**Comment by GunHog** [comment]
> Mining mechs are pretty shit though

Play on Basil when I am RD. I will happily make you a mech just to prove you wrong. We could even have a contest about it. If you die, I may brain you, make you an AI, then shove you into a Ripley and order you to mine. 

As for an Oddy, I can never remember to do it, and CMOs never ask. I will keep this in mind though!

**Comment by Repukan** [comment]
More science powercreep yay

**Comment by Repukan** [comment]
Kor how do you always bring up the fact more minerals are being delivered ( with stats even) and then turn around and in the same breath say miners aren't delivering materials.

**Comment by KorPhaeron** [comment]
Clever change thank you

@Repukan

Not that they arent, but that they might stop if there are enough fancy KA upgrades added to loot lists

**Comment by Repukan** [comment]
Most of these are gimmicks that hardly make a miner self reliant.

Is there any stats that show a lower delivery of minerals? Because I severely dislike buffing Science on a "might"

**Comment by FantasticFwoosh** [comment]
I can see this being a source of conflict with golems if they dont want to share but otherwise this is ok.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 1]
```diff
@@ -122,6 +126,13 @@
 	materials = list(MAT_METAL = 5000, MAT_GLASS = 5000, MAT_SILVER = 5000, MAT_URANIUM = 5000)
 	build_path = /obj/item/borg/upgrade/modkit/resonator_blasts
 
+/datum/design/unique_modkit/bounty
+	name = "Kinetic Accelerator Death Syphon Mod"
+	desc = "A device which causes kinetic accelerators to permenantly gain damage against creature types killed with it."
+	id = "resonatormod"
+	materials = list(MAT_GLASS = 4000, MAT_SILVER = 4000, MAT_DIAMOND = 4000, MAT_BLUESPACE = 8000)
```
Comment (GunHog) [inline code/modules/mining/lavaland/necropolis_chests.dm]: This one should totally require blood as a reagent!

**Comment by kevinz000** [comment]
It's another "I don't use item effectively therefore it's shit" episode
🤔

**Review by GunHog — APPROVED** [review]

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 2]
```diff
@@ -70,12 +70,71 @@
...
+	name = "Kinetic Accelerator Resonator Blast Mod"
+	desc = "A device which causes kinetic accelerators to fire shots that leave and detonate resonator blasts."
+	id = "resonatormod"
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 5000, MAT_SILVER = 5000, MAT_URANIUM = 5000)
+	build_path = /obj/item/borg/upgrade/modkit/resonator_blasts
+
+/datum/design/unique_modkit/bounty
+	name = "Kinetic Accelerator Death Syphon Mod"
+	desc = "A device which causes kinetic accelerators to permenantly gain damage against creature types killed with it."
+	id = "resonatormod"
```
Comment (GunHog) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Do not forget to change this, or it will conflict with the mod above!!

**Comment by Repukan** [comment]
Since it seems this will go through regarding of facts showing the fears this was created on are false, can you make it so it can only print said module only one time?

**Comment by Kromgar** [comment]
@Repukan no please i might actually use ka's now

**Comment by Repukan** [comment]
I'm talking about the special modules. It would be an extremely overpowered buff to RnD if the rapid fire module among others could be mass produced.

**Comment by ChangelingRain** [comment]
Do note that, except for the aoe mod, none of these mods stack.

**Comment by GunHog** [comment]
It is more of a nerf to miners than it is a buff to Sci. They cannot make KAs, and they cannot make these mods unless a miner brings them the correct disk.

So, instead of getting the mod directly from the chest, or buying a mod with points, the miner must buy or loot a disk, give it to Science, and ask one of them to make the mod. This costs Science minerals as well, whereas such was previously free as it did not interact with Sci at all.

The one buff here is ability to make more than one mod. As they most do not stack, it simply means that more than one miner may benefit from the same loot/purchase.

**Comment by Repukan** [comment]
Fuck off, you know very well scientists can and will get the many extra KA's in the mining base to exploit this.

It doesn't cost Science anything because they don't procure nor process the stuff. The only thing science does is press buttons and code items that allow it to bypass interactions with other departments.

All this does is putting the miner at the whims of the scientist or make his effort useless if nobody is working in RnD because everyone is too busy sticking their dicks in slimes. 

The modules are too strong for every miner to have, and should be restricted behind the RNG of the loot table and the minimum negligible amount of skill it takes to destroy a tendril. Fucking "QoL" Bullshit just makes the game easier for the lowest common denominator and powergamers.

**Comment by Repukan** [comment]
"Costs science minerals" is such a shit excuse, science doesn't even mine its shit and just because it is expected of them to bring a fucking locker and take the materials themselves now we are buffing them.

**Comment by Repukan** [comment]
But of course do tell me how it is fair to further nerf one of the roles with the highest lethality indexes so that the role with the easiest access to weapons and stuns and flying armor which gets everything in a silver fucking platter gets buffed.

**Comment by 4dplanner** [comment]
@repukan, when you say miners will get the extra KAs from the mining base to abuse this, isn't that countered by the cooldown increase from holding multiple KAs?

**Comment by Repukan** [comment]
There's ways to circumvent it pm because I'm not giving the fucking RnD exploiters more ammo.

**Comment by 4dplanner** [comment]
I'd also argue the "Death Syphon" upgrade shouldn't be tech either

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 3]
```diff
@@ -70,12 +70,71 @@
...
+//KA modkit design discs
+/obj/item/weapon/disk/design_disk/modkit_disc
+	name = "KA Mod Disk"
+	desc = "A design disc containing the design for a unique kinetic accelerator modkit."
+	icon_state = "datadisk1"
+	var/modkit_design = /datum/design/unique_modkit
 
+/obj/item/weapon/disk/design_disk/modkit_disc/Initialize()
+	. = ..()
+	var/datum/design/D = new modkit_design
```
Comment (lzimann) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Could be just `blueprints[1] = new modkit_design`, D is not really being used for anything here.

**Comment by Repukan** [comment]
Absolutely shit change

**Comment by adamkad1** [comment]
lulz, science gets something to do again

**Comment by Repukan** [comment]
That's not even the worst part of this but yeah I brought that up.

**Pull request merged**